### PR TITLE
Tweak the `doctype` check.

### PR DESCRIPTION
**Pull request by XhmikosR**
Compare it to the html5 doctype, and not just check if it's present.

It's currently untested because I'm on Windows so I need #123 plus I use FF > 46 (see #119).

### CODE REVIEW

**Comment by XhmikosR** [comment]
OK, so I see tests fail with

```
 1) Verify advice structure category: bestpractice We should return the keys for doctype.js:

     WebDriverError: unknown error: Failed to execute 'serializeToString' on 'XMLSerializer': parameter 1 is not of type 'Node'.
```

I don't see why it throws that.

**Comment by tobli** [comment]
The doctype might be null. Try moving the serialisation down.

**Comment by XhmikosR** [comment]
@tobli: indeed, that is what happens when one can't run tests :/

Now I need to fix the new test I added.

**Comment by XhmikosR** [comment]
Is it possible to make the tests run the newly added doctype4.html for a specific test? I'm not familiar with the tests process yet.

**Comment by tobli** [comment]
Absolutely! Sorry for not having documented that yet. There are a few other test that uses an extended syntax for running test, that also specifies a html file name. Browse the tests to see what you find. I can paste an example when I get the chance to sit down with the computer later in the weekend.

**Comment by XhmikosR** [comment]
No worries. My main concern is #123 and #119 because I can't really test any rule additions locally.

About the test, I have no idea how to make it work. I'll try to figure it out unless you beat me to it :)

**Comment by tobli** [comment]
Have a look at https://github.com/sitespeedio/coach/blob/master/test/dom/bestpractice/indexTest.js#L29. By passing an optional file name you can override the default (which is the same as the js file, but with an html extension instead of .js)

**Comment by XhmikosR** [comment]
Thanks, that did the job.

So, do you agree with the score I used for non-HTML5 doctype (90)?

File: test/http-server/bestpractice/doctype4.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
+<html>
+<head>
+  <meta charset="utf-8">
+  <title>Check the doctype
+  </title>
+</head>
+<body>
+  <body>
```
Comment (tobli) [inline test/http-server/bestpractice/doctype4.html]: Please remove the duplicated body tag here. The doctype.html also has this issue, it'd be great to fix that as well.

**Comment by tobli** [comment]
It'd be lovely to add a test case for missing doctype as well, to cover all cases handled by the advice.

File: test/http-server/bestpractice/doctype4.html (hunk 1 above)
Comment (XhmikosR) [inline test/http-server/bestpractice/doctype4.html]: Done. I just copied the doctype.html and change doctype hence the duplication of this error.

**Comment by XhmikosR** [comment]
@tobli; doctype.html is the case without doctype.

**Comment by tobli** [comment]
With regards to the score for non html5 doctype: my take on this is that the coach should advice you to use good practices for a modern/current web. It's not primarily a syntax checker. So with that in mind I'm tempted to say that an explicit choice to use an old doctype (and therefor parsing model) for a webpage in 2016 is worse than simply forgetting to include the doctype. My suggestion would be 25. Makes sense? Ping @soulgalore.

**Comment by tobli** [comment]
@XhmikosR Ahh of course! Sorry, missed the missing doctype test. I reverse my position totally and suggest a test case for a valid html5 doctype to give score === 100! =)

**Comment by XhmikosR** [comment]
Done. Now only the score for the non-html5 doctype is left to decide.

**Comment by XhmikosR** [comment]
I also did a small cleanup of the test files.

**Comment by XhmikosR** [comment]
While at it, I think the case for doctype not being the first is useless. I couldn't trigger this and there is no test case. Should we remove it?

**Comment by tobli** [comment]
Hi @XhmikosR!
Thanks for all your work on perfecting this rule. It'll be awesome!
You're absolutely right that the nodeType != 10 case isn't needed, since it's handled by doctype == null. There are three cases to consider:
1. No doctype at all
2. The html5 doctype
3. Another doctype

We should detect 1 and 3, and give appropriate advice. Go with DOM Level 2 api only, no need for XMLSerializer. That way the browser will normalize, and we don't need to consider case or whitespace. Looking at http://www.w3.org/TR/html5/syntax.html#the-doctype both <!DOCTYPE html> and <!DOCTYPE html SYSTEM 'about:legacy-compat'> are valid. So something like 
```document.doctype.name === 'html' && (document.doctype.systemId === '' || document.doctype.systemId === 'about:legacy-compat')```
I guess. Even though <!doctype… works, <!DOCTYPE html> is the canonical form. I'd prefer to use that in advice messages (and test files such as hars).

The other html cleanups are great. Could you please move that to a separate PR? When working with PRs it's easier if there's one focus per PR. So one for code/tests for doctype, and one for html cleanup.

Thanks!
/Tobias

**Comment by XhmikosR** [comment]
I will split this but can you please have a look at my other PRs and get
things going? It's not good for motivation to leave PRs hanging for a long
time :/

**Comment by XhmikosR** [comment]
@tobli @soulgalore: this should be ready now.

**Comment by tobli** [comment]
Hi @XhmikosR!
I understand your concern. Between full time jobs and small children we have limited time to spare. Right now our main focus is getting Browsertime stable enough for a 1.0 release, and Sitespeed.io stable and feature complete for a 4.0 release. So that’s where we spend most of the time we have, and some other stuff will not see as much activity.
Thanks
/Tobias

**Comment by XhmikosR** [comment]
@tobli: I totally understand that since I'm sort of in the same situation myself :)

That being said, you should not leave contributions without merging, especially when they are simple. Take this for instance; you have reviewed it, I have addressed every comment you made. Yet it's still not merged. Similarly with my other PRs.

**Comment by tobli** [comment]
Merging. Thanks @XhmikosR !

**Pull request merged**